.github/workflows/build_and_test.yml

```diff
@@ -120,6 +120,41 @@ jobs:
       - name: Run mypy to check types
         run: poetry run mypy .
 
+  check-submodules:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          submodules: true
+
+      # We need more history, to find the common ancestor between each REL_14_* branch.
+      # 3 months should be plenty.
+      - name: Checkout more history
+        id: checkout-deeper
+        run: git submodule foreach git fetch --shallow-since='3 months'
+
+      - name: Check vendor/postgres-v14 submodule reference
+        id: postgres-v14-submodule
+        run: |
+            cd vendor/postgres-v14
+            git fetch origin "refs/heads/REL_14_STABLE_neon:refs/heads/origin/REL_14_STABLE_neon"
+            git merge-base --is-ancestor origin/REL_14_STABLE_neon HEAD || (echo "vendor/postgres-v14 submodule is not an ancestor of the REL_14_STABLE_neon branch. Please rebase it"; exit 1)
```

Comment:
Optional, probably if-statement will have better readability (and have shorter lines)